### PR TITLE
feat: company id working

### LINKED ISSUE #38: Use company identifiers in LL_VERSION_IND
Currently a `Hex<u16>` is used to represent `CompId`, this should be an enum of company identifiers which implements `.as_u16()`.

The full list can be found here:
https://www.bluetooth.com/specifications/assigned-numbers/company-identifiers

### PR DESCRIPTION
**Pull request by chocol4te**
Closes #38

### CODE REVIEW

**Comment by chocol4te** [comment]
I know I've done something very wrong, because binary size is absolutely exploding here. `cargo run` loads 138152 bytes on this branch, versus 71148 bytes on `master`.

**Comment by jonas-schievink** [comment]
> I know I've done something very wrong, because binary size is absolutely exploding here. `cargo run` loads 138152 bytes on this branch, versus 71148 bytes on `master`.

This probably happens because we debug-print `ControlPdu`, which contains a `CompanyId`, whose `Debug` impl calls `name`, which pulls in all the name strings.

**Comment by chocol4te** [comment]
Yup, my bad, it isn't causing any issues with release builds. :)

File: src/ble/link/comp_id.rs [hunk 1]
```diff
@@ -0,0 +1,1967 @@
...
+    pub fn from_raw(raw: u16) -> Self {
+        Self(raw)
+    }
+
+    /// Get inner value from company identifier
+    pub fn as_u16(&self) -> u16 {
+        self.0
+    }
+
+    /// Get the full company name from identifer
```
Comment (jonas-schievink) [inline src/ble/link/comp_id.rs]: Please add a warning in the docs for the binary size blowup. Something like "Warning: Using this function will add about 64 KB of string data to your application!"

Comment (chocol4te) [reply src/ble/link/comp_id.rs]: Fixed :)

**Pull request merged**